src/test/java/teammates/test/cases/ui/pagedata/InstructorFeedbackResultsPageDataTest.java

```diff
@@ -0,0 +1,1584 @@
+        feedbackSessionName = dataBundle.feedbackSessions.get("session1InCourse1")
+                                             .feedbackSessionName;
+        
+        showStats = "on";
+        groupByTeam = "on";
+        sortType = "question";
+        selectedSection = InstructorFeedbackResultsPageData.ALL_SECTION_OPTION;
+        data.setBundle(logic.getFeedbackSessionResultsForInstructorFromQuestion(feedbackSessionName, courseId, 
+                                                                               instructor.email, 2));
+        
+        data.initForViewByQuestion(instructor, InstructorFeedbackResultsPageData.ALL_SECTION_OPTION, 
+                                   "on", "on");
+            
+        List<InstructorFeedbackResultsQuestionTable> singleQuestionPanelList = data.getQuestionPanels();
+        assertEquals(1, singleQuestionPanelList.size());
+        
+        InstructorFeedbackResultsQuestionTable singleQuestionPanel = singleQuestionPanelList.get(0);
+        verifyQuestionTableForQuestionView(singleQuestionPanel, dataBundle.feedbackQuestions.get("qn2InSession1InCourse1").questionMetaData.getValue());
+        
+        List<InstructorFeedbackResultsResponseRow> singleQuestionResponseRows = singleQuestionPanel.getResponses();
+        assertEquals(3, singleQuestionResponseRows.size());
+        // just verify any response
+        InstructorFeedbackResultsResponseRow singleQuestionResponseRow = singleQuestionResponseRows.get(1);
+        verifyResponseRow(singleQuestionResponseRow, dataBundle.feedbackResponses.get("response1ForQ2S1C1"),
+                          dataBundle.students.get("student2InCourse1"), dataBundle.students.get("student1InCourse1"));
+        
+        assertFalse(singleQuestionResponseRow.isGiverProfilePictureAColumn());
+        assertFalse(singleQuestionResponseRow.isRecipientProfilePictureAColumn());
+
+        assertNotNull(singleQuestionResponseRow.getModerationButton());
+        
+        modButton = singleQuestionResponseRow.getModerationButton();
+        assertEquals("Moderate Response", modButton.getButtonText());
+        verifyHtmlClass(modButton.getClassName(), "btn", "btn-default", "btn-xs");
+        
+        assertNotNull(modButton);
+        assertEquals("idOfTypicalCourse1", modButton.getCourseId());
+        assertEquals("First feedback session", modButton.getFeedbackSessionName());
+        assertEquals("student2InCourse1@gmail.tmt", modButton.getGiverIdentifier());
+        assertEquals(2, modButton.getQuestionNumber());
+        assertFalse(modButton.isDisabled());
+        
+        
+        ______TS("view by question, all sections, require loading by ajax");
+        
+        
+        feedbackSessionName = dataBundle.feedbackSessions.get("session1InCourse1").feedbackSessionName;
+        showStats = "on";
+        groupByTeam = "on";
+        sortType = "question";
+        selectedSection = InstructorFeedbackResultsPageData.ALL_SECTION_OPTION;
```

Comment:
These seem to be done relatively often.
Possible to extract into a method that returns a `*PageData`?